server.go

```diff
@@ -432,6 +432,32 @@ func (ps *peerState) connectionsWithIP(ip net.IP) int {
 	return total
 }
 
+// hostToNetAddress parses and returns an address manager network address given
+// a hostname in a supported format (IPv4, IPv6).  If the hostname cannot be
+// immediately converted from a known address format, it will be resolved using
+// a DNS lookup function.  If it cannot be resolved, an error is returned.
+//
+// This function is safe for concurrent access.
+func (cfg *config) hostToNetAddress(host string, port uint16, services wire.ServiceFlag) (*addrmgr.NetAddress, error) {
```

Comment:
If I remove this function from the config type, the test I've written breaks, and I'm struggling to figure out the right way to fix it.

Previously, in the test, I was able to overwrite the config struct from `config.go` with a new `lookup` value, like so:

```golang
for _, test := range tests {
		cfg := &config{
			lookup: test.lookupFunc,
		}
		result, err := cfg.hostToNetAddress(test.host, test.port, services)
```

But by removing this from the config type, I can no longer easily use the `test.lookupFunc`.
One way I can solve this is by just using dependency injection and adding `cfg` as a parameter for `hostToNetAddress`:

```golang
for _, test := range tests {
		cfg := &config{
			lookup: test.lookupFunc,
		}
		result, err := hostToNetAddress(cfg, test.host, test.port, services)
```

But I don't know if that's a good solution or not. It seems hacky. Do you have any suggestions?

Reply:
The most straightforward solution is to take the lookup function as a parameter to `hostToNetAddress` since that makes the dependency clear and makes it trivial to do `result, err := hostToNetAddress(cfg, test.host, test.port, services, test.lookupFunc)`.

The other callers can simply invoke it with `dcrdLookup`.  That will, imo, also have the benefit of making it quite obvious when reading the code that `hostToNetAddress` potentially performs a DNS lookup.

Reply:
You can define a type for the lookup function as well if you want.

e.g.:

```Go
type resolveIPFn func(string) ([]net.IP, error)

func hostToNetAddress(host string, port uint16, services wire.ServiceFlag, resolver resolveIPFn) (*addrmgr.NetAddress, error) {
	...
	ips, err := resolver(host)
	...
}
```

Reply:
Perfect! Thank you.
Since I've now modified the inputs for this function, I noticed that line 2297 of `server.go` redefines how `peer.go` calls `hostToNetAddress`, and it uses the old inputs. Does this PR need to expand scope to update the function in peer.go?

Reply:
I don't think so.  Just do `address, err := hostToNetAddress(host, port, services, dcrdLookup)`.

Reply:
Also, updating `peer` would make it require a new major version since it would be a breaking API change, and the purpose of taking that method to begin with is so the DNS lookups happen in the caller.

This keeps all of the resolution logic in `server.go`.

Reply:
Sounds good. This should all be resolved now.